compiler/include/misc.h

```diff
@@ -70,7 +70,10 @@
 #define USR_STOP       exitIfFatalErrorsEncountered
 
 // INT_ASSERT is intended to become no-op in production builds of compiler
-#define INT_ASSERT(x) do { if (!(x)) INT_FATAL("assertion error"); } while (0)
+#define SELECT_ASSERT(_1, _2, NAME) NAME
+#define INT_ASSERT(...) SELECT_ASSERT(__VA_ARGS__, INT_ASSERT2, INT_ASSERT1)(__VA_ARGS__)
+#define INT_ASSERT1(x) do { if (!(x)) INT_FATAL("assertion error"); } while (0)
+#define INT_ASSERT2(s, x) do { if (!(x)) INT_FATAL((s), "assertion error"); } while (0)
```

Comment:
I don't believe this will work if the two-arg form of `INT_ASSERT()` is used.  Have you tested that?

Reply:
I did not test it because I couldn't find that assertion in the tests. It seems like it's not being triggered anywhere. I think I would have to introduce it arbitrary in one of the existing tests, or create a new one, but I don't know where.

Reply:
Hi @chris980: I haven't come up with a good way to test this feature in the nightly testing (the question I was wrestling with in my previous comments), but I think for the sake of Paul's comment, inserting a call to your INT_ASSERT() somewhere into the code base where it would get called to make sure it works as intended is all that would be required (and then, having verified that it works, back that change back out again rather than committing it).

For example, compiling `chpl test/classes/ferguson/generic-inherit6.chpl --verify --devel` should hit an internal error at [line 202 of `compiler/AST/CallExpr.cpp`](https://github.com/chapel-lang/chapel/blob/master/compiler/AST/CallExpr.cpp#L201-L203).  If you replace that INT_FATAL() with an INT_ASSERT(), rebuild the compiler, and rerun the test, does your INT_ASSERT() work as intended?

Reply:
Hi @bradcray , 

Thank you for the help. 
@cassella was right, the two-arg form was not working, but I fixed it.

After modifying [`compiler/AST/CallExpr.cpp`](https://github.com/chapel-lang/chapel/blob/master/compiler/AST/CallExpr.cpp#L201-L203), it produces the expected result: `test/classes/ferguson/generic-inherit6.chpl:13: internal error: assertion error [AST/CallExpr.cpp:202]`

Reply:
@chris980: For some reason, after checking out your branch, I don't seem to be able to reproduce this.  Am I doing something wrong?  My diff to CallExpr.cpp is:

```diff
index 0fbdc24f0b..e40bc9c837 100644
--- a/compiler/AST/CallExpr.cpp
+++ b/compiler/AST/CallExpr.cpp
@@ -198,9 +198,12 @@ void CallExpr::verify() {
 
   Expr::verify(E_CallExpr);
 
+  INT_ASSERT(this, parentExpr == NULL);
+  /*
   if (parentExpr == NULL) {
     INT_FATAL(this, "Every CallExpr is expected to have a parentExpr");
   }
+  */
```

but the output I get is:

```
internal error: assertion error [AST/FnSymbol.cpp:102]
```

(note the lack of pointer to the user's source location)

Reply:
I think you mean `parentExpr != NULL`, since you're asserting it, not FATAL'ing if it's true.

The error you're getting is from a different file, I'm guessing because the assert didn't trigger so you reached different code.

Reply:
That line if that other file is an `INT_FATAL`, not an `INT_ASSERT`, so I'm guessing you converted that one the same way.  If I reverse the test on that `INT_FATAL`, the existing `INT_FATAL(this, ...)` also fails to print a user source location.  So I think the lack of a user location in that one has something to do with the `this` in that path,

Reply:
Ah, you're right, good sleuthing Paul.  I'd originally tried modifying AST/Symbol.cpp just to test something different than @chris980, didn't get the expected result, so tried the case above, but without reverting my original change.  Doing that, and fixing the polarity of my assert, I get the expected output.

(I'm still a little confused by my original test, though... I got to it by finding a test case that generated an internal error which I thought had source information associated with it; and on that one, I did think to reverse the polarity of the boolean expression.  It seems likely I got something wrong, but I'm not sure what...).